packages/gatsby-theme-carbon/src/components/GlobalSearch/GlobalSearch.js

```diff
@@ -78,6 +78,15 @@ const GlobalSearchInput = () => {
     [clearAndClose, focusedItem]
   );
 
+  useEffect(() => {
```

Comment:
@vpicone Here is my attempt to try to wrap that code in the useEffect hook. BEWARE: this code freezes the page. So do not merge it. Is this the right way to use the useEffect hook? I am relatively new to this react hook.

Reply:
Some more detail. The website freezes after the ESC key is pressed a few times on any page. The site renders fine and you are able to move around the site normally.

Reply:
You're missing the dependencies and cleanup part of the `useEffect` hook. Check out the docs here: https://reactjs.org/docs/hooks-reference.html#useeffect

Reply:
That was very helpful, by adding the cleanup part I was able to get the site to work "as expected/normally" with the additional esc functionality. 

@vpicone take a look at this one and see if it is ready to get merged.

Reply:
Doing a bit more testing... I think their still is an issue:
![image](https://user-images.githubusercontent.com/4451128/92817925-bbcd8200-f38c-11ea-9b57-81d97ab55af9.png)

This seems to happen when there is text in the search bar when esc is pressed

When there is text in the field and the esc key is pressed what would be the expected behavior? Delete the text? Or save the state?

Reply:
@aliunwala I don't think escape should have any impact on the search. If you reduce your browser width to mobile sizes, you'll see [MDN](https://developer.mozilla.org/en-US/) doesn't perform any actions there.

My hunch is it would clear the search, then if the search is clear, it would close it.